src/core/midiChannelProc.cpp

```diff
@@ -149,15 +149,34 @@ void setMute(MidiChannel* ch, bool v)
 			ch->addVstMidiEvent(MIDI_ALL_NOTES_OFF, 0);
 	#endif		
 		}
+
+	for (Channel* channel : giada::m::mixer::channels)		// This is for processing playing_inaudible
+		channel->sendMidiLstatus();
+
 	ch->sendMidiLmute();
 }
 
 
 /* -------------------------------------------------------------------------- */
 
 
+void setSolo(MidiChannel* ch, bool v)
+{
+	ch->solo = v;
+	m::mh::updateSoloCount();
+
+	for (Channel* channel : giada::m::mixer::channels)		// This is for processing playing_inaudible
```

Comment:
Is it really necessary to loop over all channels when setting solo?

Reply:
You're right, this is only necessary for setSolo. In case of setMute I'd go with:
`ch->sendMidiLstatus();`
instead.
This has to be done in both sampleChannelProc and midiChannelProc so that calls for another commit. Can I commit into pull request, or should I make a new request?